cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/ProvisionResponse.java

```diff
@@ -0,0 +1,121 @@
+  private ProvisionStatus _status;
+  private final StringBuilder _recommendation;
+
+  /**
+   * Constructor to be used for provision statuses, for which the recommendations are relevant.
+   */
+  public ProvisionResponse(ProvisionStatus status, String recommendation) {
+    this(status);
+    if (!(status == ProvisionStatus.UNDER_PROVISIONED || status == ProvisionStatus.OVER_PROVISIONED)) {
+      throw new IllegalArgumentException(String.format("Recommendation is irrelevant for provision status %s.", status));
+    }
+    validateNotNull(recommendation, "The recommendation cannot be null.");
+    _recommendation.append(recommendation);
+  }
+
+  /**
+   * Constructor to be used for provision statuses, for which the recommendations are irrelevant.
+   */
+  public ProvisionResponse(ProvisionStatus status) {
+    validateNotNull(status, "The provision status cannot be null.");
+    _status = status;
+    _recommendation = new StringBuilder();
+  }
+
+  /**
+   * @return The current provision status.
+   */
+  public ProvisionStatus status() {
+    return _status;
+  }
+
+  /**
+   * @return Recommended action regarding the current provision status.
+   */
+  public String recommendation() {
+    return _recommendation.toString();
+  }
+
+  /**
+   * Aggregate the given provision response to this provision response using the following rules: Aggregating ...
+   * <ul>
+   *   <li>any provision status with {@link ProvisionStatus#UNDER_PROVISIONED} is {@link ProvisionStatus#UNDER_PROVISIONED}.</li>
+   *   <li>a provision status {@code P} with {@link ProvisionStatus#UNDECIDED} is {@code P}.</li>
+   *   <li>{@link ProvisionStatus#RIGHT_SIZED} with {@link ProvisionStatus#RIGHT_SIZED} or {@link ProvisionStatus#OVER_PROVISIONED}
+   *   is {@link ProvisionStatus#RIGHT_SIZED}.</li>
+   *   <li>{@link ProvisionStatus#OVER_PROVISIONED} with {@link ProvisionStatus#OVER_PROVISIONED} yields itself.</li>
+   * </ul>
+   *
+   * Note that these rules enforce that once a state changes from {@link ProvisionStatus#OVER_PROVISIONED} to another state, it cannot go
+   * back to this state. Similarly, once a state goes into {@link ProvisionStatus#UNDER_PROVISIONED}, no other followup state is possible.
+   * Hence, {@link #_recommendation} for over- or under-provisioned status can be updated without losing relevant information.
+   *
+   * @param other Provision response to aggregate into this response.
+   * @return This provision response after the aggregation.
+   */
+  public ProvisionResponse aggregate(ProvisionResponse other) {
+    if (_status == ProvisionStatus.UNDER_PROVISIONED) {
+      if (other.status() == ProvisionStatus.UNDER_PROVISIONED) {
+        aggregateRecommendations(other);
+      }
+    } else {
+      switch (other.status()) {
+        case UNDER_PROVISIONED:
+          _status = ProvisionStatus.UNDER_PROVISIONED;
+          clearRecommendation();
+          aggregateRecommendations(other);
+          break;
+        case RIGHT_SIZED:
```

Comment:
This term RIGHT_SIZED is probably ok, but right-sized usually refers to laying off employees.  I've not see it used when referring to computer systems.

Reply:
Seems good then.  Thanks for checking.